### ISSUE TITLE
panic running against https://github.com/ardanlabs/service

### ISSUE THREAD

**Issue by jechols** [dominikh/go-tools#1573]
```
$ staticcheck -version
staticcheck 2023.1.7 (v0.4.7)

$ staticcheck -debug.version
staticcheck 2023.1.7 (v0.4.7)

Compiled with Go version: go1.21.0
Main module:
        honnef.co/go/tools@v0.4.7 (sum: h1:9MDAWxMoSnB6QoSqiVr7P5mtkT9pOc1kSxchzPCnqJs=)
Dependencies:
        github.com/BurntSushi/toml@v1.2.1 (sum: h1:9F2/+DoOYIOksmaJFPw1tGFy1eDnIJXg+UHjuD8lTak=)
        golang.org/x/exp/typeparams@v0.0.0-20221208152030-732eee02a75a (sum: h1:Jw5wfR+h9mnIYH+OtGT2im5wV1YGGDora5vTv/aa5bE=)
        golang.org/x/mod@v0.12.0 (sum: h1:rmsUpXtvNzj340zd98LZ4KntptpfRHwpFOHG188oHXc=)
        golang.org/x/sys@v0.11.0 (sum: h1:eG7RXZHdqOJ1i+0lgLgCpSXAp6M3LYlAo6osgSi0xOM=)
        golang.org/x/tools@v0.12.1-0.20230825192346-2191a27a6dc5 (sum: h1:Vk4mysSz+GqQK2eqgWbo4zEO89wkeAjJiFIr9bpqa8k=)

$ go version
go version go1.22.5 linux/amd64

$ go env
GO111MODULE=''
GOARCH='amd64'
GOBIN=''
GOCACHE='/home/jechols/.cache/go-build'
GOENV='/home/jechols/.config/go/env'
GOEXE=''
GOEXPERIMENT=''
GOFLAGS=''
GOHOSTARCH='amd64'
GOHOSTOS='linux'
GOINSECURE=''
GOMODCACHE='/home/jechols/.gvm/pkgsets/go1.22.5/global/pkg/mod'
GONOPROXY=''
GONOSUMDB=''
GOOS='linux'
GOPATH='/home/jechols/.gvm/pkgsets/go1.22.5/global'
GOPRIVATE=''
GOPROXY='https://proxy.golang.org,direct'
GOROOT='/home/jechols/.gvm/gos/go1.22.5'
GOSUMDB='sum.golang.org'
GOTMPDIR=''
GOTOOLCHAIN='auto'
GOTOOLDIR='/home/jechols/.gvm/gos/go1.22.5/pkg/tool/linux_amd64'
GOVCS=''
GOVERSION='go1.22.5'
GCCGO='gccgo'
GOAMD64='v1'
AR='ar'
CC='gcc'
CXX='g++'
CGO_ENABLED='1'
GOMOD='/home/jechols/projects/service/go.mod'
GOWORK=''
CGO_CFLAGS='-O2 -g'
CGO_CPPFLAGS=''
CGO_CXXFLAGS='-O2 -g'
CGO_FFLAGS='-O2 -g'
CGO_LDFLAGS='-O2 -g'
PKG_CONFIG='pkg-config'
GOGCCFLAGS='-fPIC -m64 -pthread -Wl,--no-gc-sections -fmessage-length=0 -ffile-prefix-map=/tmp/go-build903614646=/tmp/go-build -gno-record-gcc-switches'

```

Multiple people in the session today are getting panice when staticcheck runs, and Bill Kennedy said to just add a bug report real quick.

**Comment by jechols** [comment]
Reproduce by pulling the "service" project and running "make test" which runs `staticcheck -checks=all ./...`

It sounds like some people are running this fine, though. I never had staticcheck installed before so I'm wondering if it's fine with older versions or something.

**Comment by dominikh** [comment]
This is already fixed on master and in 2024.1rc1.
